exist-core/src/test/java/org/exist/xquery/ModuleImportTest.java

```diff
@@ -0,0 +1,162 @@
+ */
+package org.exist.xquery;
+
+import com.evolvedbinary.j8fu.Either;
+import org.exist.EXistException;
+import org.exist.security.PermissionDeniedException;
+import org.exist.storage.BrokerPool;
+import org.exist.storage.DBBroker;
+import org.exist.test.ExistEmbeddedServer;
+import org.exist.test.ExistXmldbEmbeddedServer;
+import org.exist.test.XQueryCompilationTest;
+import org.exist.xquery.value.Sequence;
+import org.exist.xquery.value.StringValue;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.evolvedbinary.j8fu.Either.Left;
+import static com.evolvedbinary.j8fu.Either.Right;
+import static com.ibm.icu.impl.Assert.fail;
+import static org.exist.test.XQueryAssertions.assertThatXQResult;
+import static org.exist.test.XQueryAssertions.assertXQStaticError;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Ensure library module imports work in one-off queries
+ * needs functx to be installed => conf.xml => triggers => autodeploy
+ *
+ * @author <a href="mailto:juri@existsolutions.com">Juri Leino</a>
+ */
+public class ModuleImportTest {
+    protected static Either<XPathException, CompiledXQuery> compileQuery(final String string) throws EXistException, PermissionDeniedException {
+        final BrokerPool pool = server.getBrokerPool();
+        final XQuery xqueryService = pool.getXQueryService();
+        try (final DBBroker broker = pool.getBroker()) {
+            try {
+                return Right(xqueryService.compile(new XQueryContext(broker.getDatabase()), string));
+            } catch (final XPathException e) {
+                return Left(e);
+            }
+        }
+    }
+
+    protected static Either<XPathException, Sequence> executeQuery(final String string) throws EXistException, PermissionDeniedException {
+        final BrokerPool pool = server.getBrokerPool();
+        final XQuery xqueryService = pool.getXQueryService();
+        try (final DBBroker broker = pool.getBroker()) {
+            try {
+                return Right(xqueryService.execute(broker, string, null));
+            } catch (final XPathException e) {
+                return Left(e);
+            }
+        }
+    }
+
+    private static Path getConfigFile() {
+        final ClassLoader loader = ModuleImportTest.class.getClassLoader();
+        final char separator = System.getProperty("file.separator").charAt(0);
+        final String packagePath = ModuleImportTest.class.getPackage().getName().replace('.', separator);
+
+        try {
+            return Paths.get(loader.getResource(packagePath + separator + "conf.xml").toURI());
+        } catch (final URISyntaxException e) {
+            fail(e);
+            return null;
+        }
+    }
+
+    @ClassRule
+    public static ExistEmbeddedServer server = new ExistEmbeddedServer(null, getConfigFile(), null, false, true);
```

Comment:
Please move the field declaration to the top of the file to resolve this Codacy issue...

Reply:
Will do